addons/hr_recruitment/tests/test_recruitment_allowed_user_ids.py

```diff
@@ -33,6 +33,52 @@ def setUp(self):
             'company_id': self.company_b.id,
         })
 
+        # Users with different access rights and their employees
+        self.company = self.env.company
+
+        self.manager_user = self.env['res.users'].create({
+            'name': 'Mannie the Manager',
+            'login': 'mannie',
+            'email': 'mannie@company.com',
+            'company_ids': [self.company.id],
+            'company_id': self.company.id,
```

Comment:
If you do follow the comment above then you will also have to change the company here (and in the all the following users/employees).
```suggestion
            'company_ids': [self.company_a.id],
            'company_id': self.company_a.id,
```